src/daft-core/src/array/ops/image.rs

```diff
@@ -203,6 +223,50 @@ impl<'a> DaftImageBuffer<'a> {
             _ => unimplemented!("Mode {self:?} not implemented"),
         }
     }
+
+    pub fn crop(&self, bbox: &BBox) -> Self {
+        match self {
+            DaftImageBuffer::L(imgbuf) => {
+                let imgbuf_copy = image_buffer_cow_to_vec(imgbuf);
+                let result =
+                    image::imageops::crop_imm(&imgbuf_copy, bbox.0, bbox.1, bbox.2, bbox.3)
+                        .to_image();
+                DaftImageBuffer::L(image_buffer_vec_to_cow(result))
+            }
+            DaftImageBuffer::LA(imgbuf) => {
+                let imgbuf_copy = image_buffer_cow_to_vec(imgbuf);
+                let result =
+                    image::imageops::crop_imm(&imgbuf_copy, bbox.0, bbox.1, bbox.2, bbox.3)
+                        .to_image();
+                DaftImageBuffer::LA(image_buffer_vec_to_cow(result))
+            }
+            DaftImageBuffer::RGB(imgbuf) => {
+                let imgbuf_copy = image_buffer_cow_to_vec(imgbuf);
```

Comment:
Took me a few hours to figure out the lifetimes here, but I ended up making a copy of the Cow into a Vec on the stack.

A naive `imgbuf.clone()` didn't seem to work -- for some reason the Cow's lifetime (tied to `&self`) was making the `crop_imm` function very unhappy.

Not sure why, so this is my workaround. I believe 2 copies are being made here - once in `image_buffer_cow_to_vec` and once in `.to_image()` which materializes the crop.

Reply:
The error message I get if I naively pass in `imgbuf` without copying the Cow into a Vec is:

```
borrowed data escapes outside of method
`self` escapes the method body hererustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2FUsers%2Fjaychia%2Fcode%2FDaft%2Fsrc%2Fdaft-core%2Fsrc%2Farray%2Fops%2Fimage.rs)
image.rs(227, 17): `self` is a reference that is only valid in the method body
image.rs(120, 6): lifetime `'a` defined here
```

Reply:
Hacked around this issue: https://github.com/Eventual-Inc/Daft/pull/1175/commits/ab0b606fc668105313812f29432d1a5d35edf2aa